[Wong Chuan Kai] Duke Increments

wongchuankai opened the pull request:
Duke

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

wongchuankai committed:
Level 1. Greet, Echo, Exit

wongchuankai committed:
Level 2. Add, List

wongchuankai committed:
Level 3. Mark as Done

wongchuankai committed:
Level-3 added markasdone function in task class

wongchuankai committed:
added

wongchuankai committed:
Level 4. ToDos, Events, Deadlines

wongchuankai committed:
Level 4

wongchuankai committed:
Level 5. Handle Errors

wongchuankai committed:
Level 6. Delete

wongchuankai committed:
Level-6

wongchuankai committed:
updated handle error

wongchuankai committed:
Level 7. Save

wongchuankai committed:
level -7

wongchuankai committed:
level - 7

wongchuankai committed:
Level 8. Dates and Times

wongchuankai committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.class
#	src/main/java/Duke.java

wongchuankai committed:
debug an error
 - arraylist

wongchuankai committed:
Revert "debug an error"

This reverts commit a05d7cf36cff7b5516b778f3b9ec7a9ecb3efeb1.

wongchuankai committed:
redo commit

wongchuankai committed:
Revert "redo commit"

This reverts commit 3353043b0f295ae21a5a46da16332669bacf2ad6.

wongchuankai committed:
Level 7. Save

wongchuankai committed:
Level 8. Dates and Times

wongchuankai committed:
Merge branch 'level-7-branch'

wongchuankai committed:
Merge branch 'level-8-branch'

# Conflicts:
#	src/main/java/Duke.class
#	src/main/java/Duke.java
#	src/main/java/Tasklist.txt
#	src/main/java/time.class

wongchuankai committed:
A-MoreOOP

wongchuankai committed:
JUnit Testing

wongchuankai committed:
JUnit-testing

wongchuankai committed:
added java doc comments

wongchuankai committed:
Level-9 Find

wongchuankai committed:
A-CodingStandard: Coding Standard

wongchuankai committed:
Merge branch 'branch-A-JavaDoc'

wongchuankai committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Command.java
#	src/main/java/DeleteCommand.java
#	src/main/java/DoneCommand.java
#	src/main/java/ListCommand.java
#	src/main/java/Parser.java
#	src/main/java/TaskList.java
#	src/main/java/time.java

wongchuankai committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Deadline.class
#	src/main/java/FindCommand.java
#	src/main/java/Parser.class
#	src/main/java/Parser.java
#	src/main/java/Task.class
#	src/main/java/Tasklist.txt

wongchuankai committed:
no message

wongchuankai committed:
Level-9 added javadoc comment

wongchuankai committed:
debugged Done command

wongchuankai committed:
Merge branch 'gradle'

wongchuankai committed:
add-ed gradle

wongchuankai committed:
added tutorial1 javafx

wongchuankai committed:
updated class file because I downgraded my java version to be compatible with gradle

wongchuankai committed:
added JavaFx tutorial 3

choongyx commented on src/main/java/ByeCommand.java:
```diff
@@ -0,0 +1,36 @@
+public class ByeCommand extends Command {
+
+    /**
+     *  Bye Constructor with command string input.
```
Could delete a space before "Bye" to ensure that the comments are aligned as seen in example.
![image](https://user-images.githubusercontent.com/39525563/64272845-c469cc00-cf72-11e9-87ad-9d5c47b263fa.png)

choongyx commented on src/main/java/ByeCommand.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    /**
+     * Execute the bye command
```
Header could start with "executes".
![image](https://user-images.githubusercontent.com/39525563/64272777-9b493b80-cf72-11e9-9d39-d05efa6de719.png)

choongyx commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,30 @@
+/**
+ * Execute the command in Duke.
+ */
+public class Command {
```
The command class could be an abstract class. Methods such as execute do not need to be fully described and left for the child classes to define since each of them exhibit different behaviour.

choongyx commented on src/main/java/time.java:
```diff
@@ -0,0 +1,104 @@
+/**
```
An interesting way to format the Date and Time. Could consider using the `DateFormat` class to make the formatting simpler with `DateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy HHmm")`

choongyx commented on src/main/java/DoneCommand.java:
```diff
@@ -0,0 +1,62 @@
+    @Override
+    public void execute(TaskList task, Ui ui, Storage storage) throws DukeException {
+        try {
+            String[] donearr = command.split(" ");
```
Variables should be in camelCase.

choongyx commented on src/main/java/FindCommand.java:
```diff
@@ -0,0 +1,50 @@
+        boolean none = true;
+
+        for(int i = 0; i <tasklist.size();i++) {
+            Task t = tasklist.get(i);
```
t is not an iterator, a more descriptive name could be given.

choongyx commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,46 @@
+    public Command parse(String command) {
+        if (command.startsWith("todo ")) {
+            ToDoCommand c = new ToDoCommand(command);
+            return c;
```
Could place `return c` at the end of the if-else block instead of repeating them after each case.

choongyx reviewed (commented):
The Javadocs was generally detailed and well done! Perhaps, could try checking the code by running the checkstyle with gradle for some coding style violations such as variable naming. In general, the code was easy to read. :)

shiweing commented on src/main/java/FindCommand.java:
```diff
@@ -0,0 +1,50 @@
+        ArrayList<Task> tasklist = task.getTaskList();
+
+        String[] strarr = command.split(" ");
+        String search = strarr[1];
```
You might want to check for IndexOutOfboundsException in case user does not provide a keyword.

wongchuankai committed:
Merge tag 'Level-10'

# Conflicts:
#	src/main/java/Duke.class

wongchuankai committed:
Merge tag 'Level-10'

# Conflicts:
#	src/main/java/Duke.class

wongchuankai committed:
commit

wongchuankai committed:
merged level-10

wongchuankai committed:
added javafx gui
-things to do
1) start up welcome input text
2) bye will shut off dialog tb
3) packaging
4) this week task

wongchuankai committed:
added checkstyle complete

wongchuankai committed:
added assertions

wongchuankai committed:
Merge branch 'A-Assertions' of https://github.com/wongchuankai/duke

wongchuankai committed:
Added extension C - sort

wongchuankai committed:
Added Assertions:
todo_
event_
deadline_
find_
done_
done_ _

wongchuankai committed:
assert

wongchuankai committed:
Merge branch 'A-Assertions'

# Conflicts:
#	src/main/java/DeleteCommand.class
#	src/main/java/Tasklist.txt

wongchuankai committed:
solved merge confict

wongchuankai committed:
Added code quality

wongchuankai committed:
added code quality

wongchuankai committed:
Merge pull request #6 from wongchuankai/A-CodeQuality

A-CodeQuality

wongchuankai committed:
sort changed

wongchuankai committed:
refactored.
added closing animation and disable textfield and button upon exiting.

wongchuankai committed:
edited some changes

wongchuankai committed:
edited some changes

wongchuankai committed:
Added help command.

wongchuankai committed:
Edited README.md

wongchuankai committed:
Set theme jekyll-theme-cayman

wongchuankai committed:
added Ui picture

wongchuankai committed:
Merge branch 'master' of https://github.com/wongchuankai/duke

wongchuankai committed:
Released Product JAR

wongchuankai committed:
no message

wongchuankai committed:
Edited javadoc and readME

wongchuankai committed:
Editted Readme

wongchuankai committed:
Editted readme

wongchuankai committed:
Edited ReadME

wongchuankai committed:
Updated README

wongchuankai committed:
Fixed a deadline bug

wongchuankai committed:
editted readme

wongchuankai committed:
editted readME

wongchuankai committed:
Added

wongchuankai committed:
checkstyled
